src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java

```
@@ -120,4 +134,97 @@ public void visit(AssignmentExpr assignmentExpr) {
+          String[] sampleLines = sample.split("\\r?\\n");
+          for (int i = 0; i < sampleLines.length; i++) {
+            sampleLines[i] = "* " + sampleLines[i];
+          }
+          formattedComment.append(String.join("\n", sampleLines) + "\n* </code></pre>\n");
+          break;
+        case HTML_P:
+          String p = html_p.get(0);
+          html_p.remove(0);
+          formattedComment.append("* <p> " + p + "\n");
+          break;
+        case HTML_OL:
+          List<String> ol_items = html_ol.get(0);
+          html_ol.remove(0);
+          formattedComment.append("* <ol>\n");
+          for (int i = 0; i < ol_items.size(); i++) {
+            ol_items.set(i,  "* <li>" + ol_items.get(i) + "\n");
+          }
+          formattedComment.append(String.join("",ol_items) + "* </ol>\n");
+          break;
+        case HTML_UL:
+          List<String> ul_items = html_ul.get(0);
+          html_ul.remove(0);
+          formattedComment.append("* <ul>\n");
+          for (int i = 0; i < ul_items.size(); i++) {
+            ul_items.set(i,  "* <li>" + ul_items.get(i) + "\n");
+          }
+          formattedComment.append(String.join("",ul_items) + "* </ul>\n");
+          break;
+      }
+    }
+    for (String name : paramList.keySet()) {
+      formattedComment.append("* @param " + name + " " + paramList.get(name) + "\n");
+    }
+    if (javaDocComment.deprecated().isPresent()) {
+      formattedComment.append("* @deprecated " + javaDocComment.deprecated() + "\n");
+    }
+    if (javaDocComment.throwsText().isPresent()) {
+      formattedComment.append("* @throws " + javaDocComment.throwsText() + "\n");
+    }
+    String formattedSource = format(formattedComment.append("*/").toString());
+    return formattedSource;
+  }
+
+  public String format(String comment) {
+    String formattedComment = "";
+    try{
+      formattedComment = new Formatter().formatSource(comment);
+    }catch(Exception e) {
```

What if we just let the formatter throw the exception? We don't need to handle it for the end-user, since the failure is probably legit.

---

cool, added exception to the methods.